app/V1Module/presenters/ExerciseFilesPresenter.php

```diff
@@ -194,6 +200,10 @@ public function actionDeleteSupplementaryFile(string $id, string $fileId) {
     $exercise->updatedNow();
     $exercise->removeSupplementaryEvaluationFile($file);
     $this->exercises->flush();
+
+    $this->configChecker->check($exercise);
+    $this->exercises->flush();
```

Comment:
Is it really necessary flush exercises twice?

Reply:
I'm not 100% sure about this either. I think that setting the `broken` flag after validation should occur in a separate transaction so that it doesn't interfere with removing the file - that is a different operation. On the other hand, you could argue that if validation fails, the file should not be removed either. Any thoughts?

Reply:
Yeah, I though so (validator seeing changes).
As for the first thought... if we are setting broken flag, then in my opinion removing the file anyway is correct action to do.